src/reactA11yAccessibleHeadingsRule.ts

```diff
@@ -0,0 +1,164 @@
+import * as ts from 'typescript';
+import * as Lint from 'tslint';
+
+import {ExtendedMetadata} from './utils/ExtendedMetadata';
+
+const BAD_ORDER_HEADING_FAILURE_STRING: string = 'Heading elements should be used for structuring information on the page';
+const EMPTY_HEADING_FAILURE_STRING: string = 'Heading elements must not be empty';
+const BAD_HEADING_LENGTH_STRING: string = 'Heading content should be concise';
+const H1_HEADING_NOT_SUBSET_OF_TITLE_STRING: string = 'H1 heading should match a subset of the words in the title element';
+const BAD_NUMBER_H1_HEADING_FAILURE_STRING: string = 'H1 heading cannot exceed 2 elements';
+const VALID_HEADING_TYPES: string[] = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'];
```

Comment:
> `VALID_HEADING_TYPES: string[]`

Turn this into a `Set` so you can use `.has(` later on instead of a more complex `.find`.